x/budget/types/utils.go

```diff
@@ -18,3 +22,14 @@ func MustParseRFC3339(s string) time.Time {
 func DateRangesOverlap(startTimeA, endTimeA, startTimeB, endTimeB time.Time) bool {
 	return startTimeA.Before(endTimeB) && endTimeA.After(startTimeB)
 }
+
+func DeriveAddress(addressType AddressType, moduleName, name string) sdk.AccAddress {
+	switch addressType {
+	case AddressType32Bytes:
+		return sdk.AccAddress(address.Module(moduleName, []byte(name)))
+	case AddressType20Bytes:
+		return sdk.AccAddress(crypto.AddressHash([]byte(moduleName + name)))
+	default:
+		return sdk.AccAddress{}
```

Comment:
How about we throw invalid type here?

Reply:
Rather than returning the error, exceptions were made as follows.
https://github.com/tendermint/budget/blob/aea1ac4b4fb6996cf4b845236a5282c50a07b9a5/x/budget/keeper/grpc_query.go#L84-L86